dart/README.md

```diff
@@ -1,2 +1,218 @@
-A sample command-line application with an entrypoint in `bin/`, library code
-in `lib/`, and example unit test in `test/`.
+# Fury Dart
+
+## Overview
+
+This PR adds Dart language support to Apache Fury, implementing a comprehensive serialization solution for Dart and Flutter applications. Fury Dart consists of approximately 15,000 lines of code and provides an efficient serialization mechanism that works within Flutter's reflection limitations.
+
+## Implementation Approach
+
+Dart supports reflection, but Flutter explicitly prohibits it. To address this constraint, Fury Dart uses a combination of:
+
+1. Core serialization/deserialization logic
+2. Static code generation for type handling
+
+This approach ensures compatibility with Flutter while maintaining the performance and flexibility expected from Fury.
+
+## Features
+
+- XLANG mode support for cross-language serialization
+- Reference tracking for handling object graphs
+- Support for primitive types, collections, and custom classes
+- Serializer registration system
+- Code generation for class and enum serialization
+- Support for nested collections with automatic generic type conversion
+- Custom serializer registration
+- Support for using ByteWriter/ByteReader as serialization sources
+
+## Usage Examples
+
+### Basic Class Serialization
+
+```dart
+import 'package:fury_core/fury_core.dart';
+
+part 'some_class.g.dart';
+
+@furyClass
+class SomeClass with _$SomeClassFury {
+  late int id;
+  late String name;
+  late Map<String, double> map;
+
+  SomeClass(this.id, this.name, this.map);
+
+  SomeClass.noArgs();
+}
+```
+
+After annotating your class with `@furyClass`, run:
+
+```bash
+dart run build_runner build
+```
+
+This generates the necessary code in `some_class.g.dart` and creates the `_$SomeClassFury` mixin.
+
+### Serializing and Deserializing
+
+```dart
+Fury fury = Fury(
+  xlangMode: true,
```

Comment:
Support native serialization and xlang serialization both can be very complicated. Since the type system of xlang and native can be very different. Will xlang mode cover 95% serialization scenerios in dart? If so, we can just drop native serialization support

Reply:
I'm also OK with both native serialization and xlang serialization, it makes the implementation more versatile. Just a little concern about the work when we need to update the proposal in the future.

Reply:
Hi @LouisLou2 , thank you for the comprehensive response and insights. If we decide to focus on dart XLANG mode, we could remove xlang mode related code and document in this pr